src/binding_model.rs

```diff
@@ -1,6 +1,76 @@
 use hal;
 
+use {BindGroupLayoutHandle, BufferHandle, SamplerHandle, TextureViewHandle};
+
+bitflags! {
+    #[repr(transparent)]
+    pub struct ShaderStageFlags: u32 {
+        const NONE = 0;
+        const VERTEX = 1;
+        const FRAGMENT = 2;
+        const COMPUTE = 4;
+    }
+}
+
+#[repr(C)]
+pub enum BindingType {    
+    UniformBuffer = 0,
+    Sampler = 1,
+    SampledTexture = 2,
+    StorageBuffer = 3,
+}
+
+#[repr(C)]
+pub struct BindGroupLayoutBinding {
+    pub binding: u32,
+    pub visibility: ShaderStageFlags,
+    pub ty: BindingType,
+}
+
+#[repr(C)]
+pub struct BindGroupLayoutDescriptor<'a> {
+    pub bindings: &'a [BindGroupLayoutBinding],
```

Comment:
is this reference guaranteed to be just `*BindGroupLayoutBinding` when calling from C?